server/service/src/sync/translations/sensor.rs

```diff
@@ -0,0 +1,195 @@
+use crate::sync::{
+    api::RemoteSyncRecordV5,
+    sync_serde::{
+        date_option_to_isostring, naive_time, zero_date_as_option, empty_str_as_option_string,
+    },
+};
+use chrono::{NaiveDate, NaiveDateTime, NaiveTime};
+
+use repository::{
+    SensorRow, SensorRowRepository, ChangelogRow, ChangelogTableName, StorageConnection,
+    SyncBufferRow,
+};
+use serde::{Deserialize, Serialize};
+
+use super::{
+    IntegrationRecords, LegacyTableName, PullDependency, PullUpsertRecord, SyncTranslation,
+};
+
+const LEGACY_TABLE_NAME: &'static str = LegacyTableName::SENSOR;
+
+fn match_pull_table(sync_record: &SyncBufferRow) -> bool {
+    sync_record.table_name == LEGACY_TABLE_NAME
+}
+fn match_push_table(changelog: &ChangelogRow) -> bool {
+    changelog.table_name == ChangelogTableName::Sensor
+}
+
+#[allow(non_snake_case)]
+#[derive(Deserialize, Serialize)]
+pub struct LegacySensorRow {
+    #[serde(rename = "ID")]
+    pub id: String,
+    #[serde(rename = "name")]
+    pub name: String,
+    #[serde(rename = "macAddress")]
+    pub serial: String,
+    #[serde(rename = "locationID")]
+    #[serde(deserialize_with = "empty_str_as_option_string")]
+    pub location_id: Option<String>,
+    #[serde(rename = "storeID")]
+    #[serde(deserialize_with = "empty_str_as_option_string")]
+    pub store_id: Option<String>,
+    #[serde(rename = "batteryLevel")]
+    pub battery_level: Option<i32>,
+    #[serde(rename = "logInterval")]
+    pub log_interval: Option<i32>,
+    pub is_active: bool,
+    #[serde(rename = "lastConnectionDate")]
+    #[serde(deserialize_with = "zero_date_as_option")]
+    #[serde(serialize_with = "date_option_to_isostring")]
+    pub last_connection_date: Option<NaiveDate>,
+    #[serde(rename = "lastConnectionTime")]
+    #[serde(deserialize_with = "naive_time")]
+    pub last_connection_time: NaiveTime,

```

Comment:
Hmm, i have to remember why date and time differ (one is optional), will try to get back to you on that one

Reply:
Yeah, wasn't sure about that either - just did it the same way as `[transact]confirm_date/time`

Reply:
> will try to get back to you on that one

Added to my list of TODO, won't quite look at it now